jupyterlab_comments/jupyterlab_comments/git_commands.py

```diff
@@ -93,8 +106,13 @@ def get_code_review_comments(self, file_path_from_repo_root, git_root_dir):
 		return review_json
 
 
-	def add_comment(self, file_path, server_root):
-		pass
+	def add_detached_comment(self, file_path_from_repo_root, git_root_dir, comment_string):
+		self.run(git_root_dir, 'appraise', 'comment', '-d', '-m', comment_string, '-f', file_path_from_repo_root)
+		self.push_local_comments(git_root_dir)
```

Comment:
You're calling both `push_local_comments` here, and calling `appraise_push` in the `AddDetachedCommentsHandler.push` method after it invokes this method.

That means there is a redundant push, and one of them can be eliminated.